beacon-chain/core/state/state.go

```diff
@@ -44,6 +45,11 @@ func InitialBeaconState(
 		latestBlockRoots[i] = params.BeaconConfig().ZeroHash[:]
 	}
 
+	latestBalances := make([]uint64, len(genesisValidatorRegistry))
```

Comment:
This is not part of the spec under `get_initial_beacon_state `. See https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#on-startup. I'd rather keep `InitialBeaconState` generic and not work with `genesis.json` which will be deprecated super soon, otherwise, it feels kinda hacky. Can we just pass the values of `genesis.json` (pub key and balances) inside `initialValidatorDeposits` and use that instead? It will still register via `v.ProcessDeposit`

Reply:
If we are going to deprecate genesis.json super soon, I rather do it in this PR then and have a clean break. Also i guess `initialValidators` is not used anywhere so I can start removing that too

Reply:
Exactly. So outside of InitialBeaconState, we should do something like:
 
If demo config:
   Convert Genesis.json to initialValidatorDeposit 

Lmk if that doesn't make sense

Reply:
Reason being genesisValidatorRegistry will always be empty when we don't use Genesis.json